Mdicontainer iconchanges

gero3 opened the pull request:
<!-- Please read CONTRIBUTING.md before submitting a pull request -->

Fixes #4650


## Proposed changes

- cleanup code in accordance to @RussKie review in #4650

<!-- We are in TELL-MODE the following section must be completed -->

## Customer Impact

- None

## Regression? 

- No

## Risk

- None

<!-- end TELL-MODE -->


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4664)

dreddy-work commented on src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:
```diff
@@ -96,7 +94,7 @@ internal MenuStrip MergedMenu
 
-        private Image GetTargetWindowIcon()
+        private (Image Image,bool Visible) GetTargetWindowIcon()
         {
```
@RussKie , was comment about not querying for Icon incase it was not set visible?

gero3 replied:
I've made it so it doesn't load/ generate the image if the image isn't visible in the next commit.

dreddy-work commented on src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:
```diff
@@ -105,7 +103,14 @@ private Image GetTargetWindowIcon()
+            _system.Image = imageInfo.Image;
+            _system.Visible = imageInfo.Visible;
         }
         private bool GetTargetWindowIconVisibility() => _target is not Form formTarget || formTarget.ShowIcon;
```
nit: newline missing.

dreddy-work commented on src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:
```diff
@@ -150,9 +155,8 @@ private void OnTargetWindowHandleRecreated(object sender, EventArgs e)
                 _system.DropDown.Items.Clear();
                 _system.DropDown.Dispose();
             }
+            UpdateIcon();
```
nit: enforcing empty newline after closing braces in the repo. If existing code doesn't follow, please feel free to make those cleanup changes as well.

gero3 commented:
@dreddy-work 

I've updated the PR with the remarks you gave me and I've also updated the `UpdateIcon` method to make sure that the previous image is disposed of.

dreddy-work commented on src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:
```diff
@@ -96,23 +94,33 @@ internal MenuStrip MergedMenu
         }
 
-        private Image GetTargetWindowIcon()
+        private (Image Image,bool Visible) GetTargetWindowIcon()
```
nit: missing format on this line.

dreddy-work commented on src/System.Windows.Forms/src/System/Windows/Forms/MDIControlStrip.cs:
```diff
@@ -96,23 +94,33 @@ internal MenuStrip MergedMenu
-            _system.Image = GetTargetWindowIcon();
-            _system.Visible = GetTargetWindowIconVisibility();
+            var imageInfo = GetTargetWindowIcon();
+            if (_system.Image is not null) {
```
nit: braces always start on newline and a newline after closing braces

dreddy-work reviewed (commented):
logically nothing should change but can you please do a integration test and confirm result with gif?

RussKie commented:
Please rebase, resolve the conflicts, and address the feedback.

dreddy-work commented:
@gero3, just want to make sure you are still on this.

dreddy-work commented:
closing as there were no recent activity on this.
